microservices/cmslitemetadata_to_redshift/cmslitemetadata_to_redshift.py

```diff
@@ -32,408 +32,413 @@
+        if 'rename' in data:
+            for thisfield in data['rename']:
+                if thisfield['old'] in _df.columns:
+                    _df.rename(columns={thisfield['old']: thisfield['new']},
+                               inplace=True)
+
+        # Run replace on some fields to clean the data up
+        if 'replace' in data:
+            for thisfield in data['replace']:
+                _df[thisfield['field']].str.replace(thisfield['old'],
+                                                    thisfield['new'])
+
+        # Clean up date fields, for each field listed in the dateformat array
+        # named "field" apply "format". Leaves null entries as blanks instead
+        # of NaT.
+        if 'dateformat' in data:
+            for thisfield in data['dateformat']:
+                _df[thisfield['field']] = pd.to_datetime(
+                    _df[thisfield['field']]).apply(
+                        lambda x: x.strftime(
+                            thisfield['format']) if not pd.isnull(x) else '')
+
+        # We loop over the columns listed in the JSON configuration file.
+        # There are three sets of values that should match to consider:
+        # - columns_lookup
+        # - dbtables_dictionaries
+        # - dbtables_metadata
+
+        # The table is built in the same way as the others, but this allows us
+        # to resuse the code below in the loop to write the batch file and run
+        # the SQL command.
+
+        dictionary_dfs = {}  # keep the dictionaries in storage
+        # loop starts at index -1 to process the main metadata table.
+
+        # build an aggregate query which will be used to make one transaction
+        copy_queries = {}
+        for i in range(-1, len(columns_lookup)*2):
+            # the metadata table is built once
+            if i == -1:
+                column = "metadata"
+                dbtable = "metadata"
+                key = None
+                columnlist = columns_metadata
+                df_new = _df.copy()
+            # the column lookup tables are built
+            elif i < len(columns_lookup):
+                key = "key"
+                column = columns_lookup[i]
+                columnlist = [columns_lookup[i]]
+                dbtable = dbtables_dictionaries[i]
+                df_new = to_dict(_df, column)  # make dict a df of this column
+                dictionary_dfs[columns_lookup[i]] = df_new
+            # the metadata tables are built
+            else:
+                i_off = i - len(columns_lookup)
+                key = None
+                column = columns_lookup[i_off]
+                columnlist = ['node_id', 'lookup_id']
+                dbtable = dbtables_metadata[i_off]
+
+                # retrieve the dict in mem
+                df_dictionary = dictionary_dfs[column]
+
+                # for each row in df
+                df_new = pd.DataFrame(columns=columnlist)
+                for index, row in _df.copy().iterrows():
```

Comment:
iterrows() yields a (index, Series) tuple object as it iterates over the DataFrame. In this line we unpack that tuple into `index`, and `row`. Since we don't use `index` after unpacking it, we are getting a Pylint `[unused-variable] Unused variable 'index'` warning.

I propose changing the yield to a single object (e.g., `iterrows_tuple`) and then in the first line of the loop extract only the `row` from the iterrows_tuple object, as in:

```
for index, row in _df.copy().iterrows():
    row = iterrows_tuple[1]
```

This resolves the Pylint `[unused-variable]` warning. I will make this change.